AgentDiscoveries-Backend/src/main/java/org/softwire/training/api/routes/v1/TokenRoutes.java

```diff
@@ -62,6 +62,11 @@ public TokenResponseApiModel createToken(Request req, Response res) {
     private TokenResponseApiModel generateToken(User user) {
         // Use the user_id as the subject for the issued token
         TokenIssuer.IssuedToken issuedToken = tokenIssuer.generateToken(Integer.toString(user.getUserId()));
-        return new TokenResponseApiModel(issuedToken.getToken(), issuedToken.getExpiryInstant().toString(), user.getUserId(), user.isAdmin());
+
+        if (user.getAgentId() == null) {
+            user.setAgentId(-1);
```

Comment:
Rather than setting the agent id, can we just set a temporary variable that we access in the constructor below?
E.g. use
`int agentId = user.getAgentId() == null ? -1 : user.getAgentId();`

Reply:
Although, I'm also quite uncomfortable with using -1 as a magic value (e.g. one the front end has to know about too)
Could we actually just set a boolean field isAgent on the token response model and then this will be much more natural?

Reply:
All done